src/components/content/order/formElement/Common.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+import { ChangeEvent } from 'react';
+import { CheckboxChangeEvent } from 'antd/lib/checkbox';
+import { Ocl, Region, RegisterServiceEntity } from '../../../../xpanse-api/generated';
+import { Area } from '../../../utils/Area';
+
+export type TextInputEventHandler = (event: ChangeEvent<HTMLInputElement>) => void;
+export type NumberInputEventHandler = (value: number | string | null) => void;
+export type CheckBoxOnChangeHandler = (e: CheckboxChangeEvent) => void;
+export type SwitchOnChangeHandler = (checked: boolean) => void;
+export type ParamOnChangeHandler =
+    | TextInputEventHandler
+    | NumberInputEventHandler
+    | CheckBoxOnChangeHandler
+    | SwitchOnChangeHandler;
+
+export interface DeployParam {
+    name: string;
+    kind: string;
+    type: string;
+    example: string;
+    description: string;
+    value: string;
+    mandatory: boolean;
+    validator: string;
+}
+
+export interface DeployParamItem {
+    item: DeployParam;
+
+    [key: string]: any;
+}
+
+export function listToMap(list: any[], key: string): Map<string, any[]> {
+    let map: Map<string, any[]> = new Map<string, any[]>();
+    list.map((val) => {

```

Comment:
map function of an array is needed to create a new list. If we want to loop through an array, then i think "forEach" is better.